speculos/mcu/seproxyhal.py

```diff
@@ -48,23 +48,61 @@ class SephTag(IntEnum):
 RENDER_METHOD = RenderMethods(0, 1)
 
 
-def ticker(add_tick):
-    """
-    Function ran in a thread to trigger a tick at a periodic interval.
+class TimeTickerDaemon(threading.Thread):
+    def __init__(self, add_tick, *args, **kwargs):
+        super(TimeTickerDaemon, self).__init__(name="time_ticker", daemon=True)
+        self.stop: bool = False
+        self.pause: bool = False
+        self.resume: threading.Condition = threading.Condition()
+        self.add_tick: Callable = add_tick
 
-    There is no need to prevent clock drift.
-    """
+        """
+        Initializes the Ticker Daemon
+        The goal of this daemon is to emulate the time spent in the application by sending it
+        ticker events at a regular interval TICKER_DELAY.
+        This daemon can be paused and resumed through it's API to stop the flow of time in the MCU
+
+        :param add_tick: The callback function to add a Ticker Event to the packet manager
+        :type add_tick: Backend
+        """
```

Comment:
Unconventional, to have this doc block after the method. :thinking:

Reply:
Done